api/services_cluster.go

```diff
@@ -0,0 +1,114 @@
+package api
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/microcloud/microcloud/api/types"
+	"github.com/canonical/microcloud/microcloud/service"
+)
+
+// ServicesClusterCmd represents the /1.0/services/cluster/{name} API on MicroCloud.
+var ServicesClusterCmd = func(sh *service.Handler) rest.Endpoint {
+	return rest.Endpoint{
+		AllowedBeforeInit: true,
+		Name:              "services/cluster/{name}",
+		Path:              "services/cluster/{name}",
+
+		Delete: rest.EndpointAction{Handler: authHandler(sh, removeClusterMember), AllowUntrusted: true},
+	}
+}
+
+// removeClusterMember removes the given cluster member from all services that it exists in.
+func removeClusterMember(state *state.State, r *http.Request) response.Response {
+	force := r.URL.Query().Get("force") == "1"
+	name, err := url.PathUnescape(mux.Vars(r)["name"])
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	allServices := map[types.ServiceType]string{
+		types.MicroOVN:  MicroOVNDir,
+		types.MicroCeph: MicroCephDir,
+		types.LXD:       LXDDir,
+	}
+
+	existingServices := []types.ServiceType{types.MicroCloud}
+	for serviceType, stateDir := range allServices {
+		if service.Exists(serviceType, stateDir) {
+			existingServices = append(existingServices, serviceType)
+		}
+	}
+
+	addr, _, err := net.SplitHostPort(state.Address().URL.Host)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("State address %v is invalid: %w", state.Address().String(), err))
+	}
+
+	sh, err := service.NewHandler(state.Name(), addr, state.OS.StateDir, false, false, existingServices...)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Remove the node from LXD first as it depends on the others.
+	var memberExists bool
+	lxd := sh.Services[types.LXD]
+	existingMembers, err := lxd.ClusterMembers(r.Context())
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// The cluster member may not exist for this service if it was removed manually, so skip removal for that service.
+	_, ok := existingMembers[name]
+	if !ok {
+		logger.Warn("Cluster member not found for service", logger.Ctx{"service": lxd.Type(), "member": name})
+	} else {
+		memberExists = ok
+		err = lxd.DeleteClusterMember(r.Context(), name, force)
+		if err != nil {
+			return response.SmartError(err)
+		}
+	}
+
+	err = sh.RunConcurrent(false, true, func(s service.Service) error {
```

Comment:
Should we allow a member to be removed from the `microcloudd` cluster if microceph and microovn may fail to remove? I forsee confusion if `microcloud cluster list` reflects that the cluster member was removed but the member is still present in the Micro{Ceph,OVN} clusters (e.g. user only looks at the `cluster list` from microcloud and never checks the Ceph/OVN ones). Ofc this endpoint will still work to remove a member from the other service's clusters even if its not present in the microcloudd cluster, but that may not be clear.

Reply:
Yeah that's a good point, we should remove from LXD first and MicroCloud last.

I was considering making the whole thing sequential and removing the node from MicroCeph second since it has the most restrictions, particularly since it requires `--force` if the remaining cluster size will be less than 3.

But thinking about it more, I don't necessarily think MicroOVN should require `--force` just because MicroCeph requires it, so I think we should still try to cleanly remove the node from MicroOVN concurrently with MicroCeph. 

So If removing a node from a 3-node or smaller cluster, we will always require `--force` to be specified both for MicroCeph and necessarily also MicroCloud since it's tried last.